Submitting solution and tests for branch nc-news-6 (Ticket 7)

AnnaTas77 opened the pull request.

AnnaTas77 committed:
Submitting solution and tests for branch nc-news-6 (Ticket 7)

EddTheDuck commented on __tests__/app.test.js:
```diff
@@ -196,3 +196,59 @@ describe("GET /api/articles/:article_id/comments", () => {
+            .then(({ body }) => {
+                const { postedComment } = body;
+                // console.log("postedComment =>>", postedComment);
+
```
Please remove console.logs / commented out code

EddTheDuck commented on __tests__/app.test.js:
```diff
@@ -196,3 +196,59 @@ describe("GET /api/articles/:article_id/comments", () => {
+                expect(body.msg).toBe("Bad request");
+            });
+    });
+});
```
These are good tests but you shoudl also test for the following - 
Status 201, ignores unnecessary properties
Status 404, non existent ID, e.g. 0 or 9999
Status 404, username does not exist

EddTheDuck reviewed (changes requested):
Please add the tests mention in my comment and resubmit, thanks

AnnaTas77 committed:
Committing the requested changes on the test suite
Created 'users.model.js' and a function that checks if the provided username exists in the DB

Dutton1990 reviewed (approved):
Super refactor Anna, thanks for adding those new tests. Looks great to me you can merge and continue!
